dcos/dcos-ui
# DCOS-50271: Check for labels with duplicate keys

Add validation to jobs form to check that the user hasn't tried to add multiple labels with the same key.

Closes DCOS-50271.

## Testing

- Go to jobs > create
- In the first tab, add an id, command ("test", "sleep 100")
- Go to run config tab
- Add a label with "a" for key and "b" for value
- Add another label with "a" for key and "c" for value. You should see this
<img width="1252" alt="Screen Shot 2019-03-15 at 4 19 42 PM" src="https://user-images.githubusercontent.com/19582796/54466750-82d43300-473e-11e9-8628-d3b57b5d5cbd.png">

- Click submit, you should see errors in three places (banner in form tab, the label inputs, and in the JSON editor)
<img width="484" alt="Screen Shot 2019-03-15 at 4 20 03 PM" src="https://user-images.githubusercontent.com/19582796/54466772-997a8a00-473e-11e9-8974-122d603f257b.png">
<img width="481" alt="Screen Shot 2019-03-15 at 4 20 12 PM" src="https://user-images.githubusercontent.com/19582796/54466777-a13a2e80-473e-11e9-9043-52042a21c6a1.png">
<img width="612" alt="Screen Shot 2019-03-15 at 4 20 20 PM" src="https://user-images.githubusercontent.com/19582796/54466789-a5664c00-473e-11e9-9b4a-df7829451b8f.png">


## Trade-offs

Need to validate labels separately from the rest of the validation since they are stored in a different format in the job spec (tracked internally) compared to the job output (what's in JSON editor).

## Dependencies

N/A
